src/components/Reports/GDPRPermissions/PermissionExpandableRow.js

```diff
@@ -17,29 +18,53 @@ const PermissionValue = ({ value }) => {
   );
 };
 
-const PermissionExpandableRow = ({ field, dependencies: { experimentalFeatures } }) => {
+const PermissionExpandableRow = ({
+  field,
+  email,
+  dependencies: { dopplerApiClient, experimentalFeatures },
+}) => {
   const [expanded, setExpanded] = useState(false);
+  const [reload, setReload] = useState(false);
+  const [loading, setLoading] = useState(true);
+  const [permissions, setPermissions] = useState([]);
+  const [unexpectedError, setUnexpectedError] = useState(false);
   const intl = useIntl();
   const _ = (id, values) => intl.formatMessage({ id }, values);
   const isPermissionHistoryEnabled =
     experimentalFeatures && experimentalFeatures.getFeature('PermissionHistory');
 
-  /**
-   * Toggle the expanded state of the row and fetch the permissions if the row is expanded.
-   */
-  const toggleExpanded = () => {
-    setExpanded(!expanded);
+  const fetchData = async () => {
+    if (isPermissionHistoryEnabled) {
+      setLoading(true);
+      const fieldName = field.name;
+      const { success, value } = await dopplerApiClient.getSubscriberPermissionHistory({
+        email,
+        fieldName,
+      });
+      if (success) {
+        setPermissions(value.items);
+      } else {
+        setUnexpectedError(true);
+      }
```

Comment:
Nunca se vuelve a setear `unexpectedError` en `false`, creo que eso generaría un comportamiento incorrecto en el caso del reintento.

Sería interesante programar un test para reproducir el escenario, confirmar si funciona mal como sospecho, y luego arreglarlo y que el test pase.

Creo que un posible fix sería:

```suggestion
      if (success) {
        setPermissions(value.items);
        setUnexpectedError(false);
      } else {
          setPermissions([]);
          setUnexpectedError(true);
      }
```

Aunque mejor todavía sería usar un solo `useReducer()` (o, tal vez, un `useState()` con un estado compuesto) con un  en lugar de tantos `useState()` atómicos.

Reply:
Coincido con Andrés, y un test de esto vendría genial.

Reply:
Ya no se se incluye la opción de reintentar. También me parece buena idea el test.